src/Symfony/Component/Security/Http/Authentication/DefaultAuthenticationSuccessHandler.php

```diff
@@ -75,17 +77,39 @@ public function setOptions(array $options)
      * Get the provider key.
      *
      * @return string
+     *
+     * @deprecated since 5.2, use getFirewallName() instead
      */
     public function getProviderKey()
     {
-        return $this->providerKey;
+        if (1 !== \func_num_args() || true === func_get_arg(0)) {
+            trigger_deprecation('symfony/security-core', '5.2', 'Method "%s" is deprecated, use "getFirewallName()" instead.', __METHOD__);
+        }
+
```

Comment:
What about adding the deprecation message for the protected property here? If `$this->providerKey` is not `null`, if probably means that it was overridden by a class extending this one.

Reply:
I've added a deprecation notice and tried to fix this a bit more. I think it's BC now.